riverdriver/riverpgxv5/river_pgx_v5_driver.go

```diff
@@ -475,14 +475,10 @@ func (l *Listener) Close(ctx context.Context) error {
 		return nil
 	}
 
-	err := l.conn.Conn().Close(ctx)
-
-	// Regardless of the error state returned above, always release and unset
-	// the listener's local connection.
```

Comment:
I'm not actually sure this is a good idea and I don't think we want to return the conn to the pool. If there are `LISTEN` subscriptions active which weren't properly `UNLISTEN`'d, the conn may still receive additional notifications that could interfere with other operations on it.

For pgbouncer compatibility reasons, we will probably also end up going a direction where the listener conn is created from a standalone config or at least a separate pool from the main river db pool.

Or maybe you can elaborate more on why you think this is a necessary change?

Reply:
Okay, fair point on that one, I forgot about the potential taint from not correctly using `Unlisten`. I put the `Close` back in and added a comment to remind us why it's written this way. One of the test cases had to go because pgx's implementation is _very_ specific, and `AfterRelease` doesn't get called if a connection was released that was already closed.

Want to take another look?